geonode/base/utils.py

```diff
@@ -218,3 +218,11 @@ def _restore_owner_permissions(self):
                     assign_perm(perm, self.resource.owner, self.resource)
                 elif perm not in {'change_resourcebase_permissions', 'publish_resourcebase'}:
                     assign_perm(perm, self.resource.owner, self.resource)
+
+
+def add_keywords_to_resource(resource, keywords):
+    for keyword in keywords:
+        resource.keywords.add(keyword)
+
+    resource.save()
```

Comment:
This method seems to be used only for testing purposes.
Please do not mix utility methods for testing with the code on the core.
Remove it from `base` and place it on some other specific utility module for testing.

Other than this, `resource.save()` is not needed here.

Reply:
This can be a utility method for `ResourceBase`, as we are passing in Resource to the method, and the functionality has forseeable usage. But if agreed to not use for that, then we can move it `populate_test_data` itself.

Reply:
Yes, please remove it.

Keep in mind that in GeoNode modifying the models directly is always highly discouraged and moreover invoking a `save()` on them.

This is something that can be done during the test phase, but not in the core. Why? Because the `'save()` methods in GeoNode are linked to a bunch of signals that interact with the GIS backend. They are not only heavy and time-consuming, but also intrinsically dangerous.

This is why we should not have utilities on the core that invoke a `save()` call.